clients/fides-js/src/fides-tcf.ts

```diff
@@ -125,41 +125,39 @@ const updateCookieAndExperience = async ({
   cookie,
   experience,
   debug = false,
-  hasValidFidesStringOverride,
   isExperienceClientSideFetched,
 }: {
   cookie: FidesCookie;
   experience: PrivacyExperience;
   debug?: boolean;
-  hasValidFidesStringOverride: boolean;
   isExperienceClientSideFetched: boolean;
 }): Promise<{
   cookie: FidesCookie;
   experience: Partial<PrivacyExperience>;
 }> => {
-  // If string override exists and is valid, the cookie has already been overridden
-  if (hasValidFidesStringOverride) {
-    // However, if we didn't get the experience until the client side fetch, we need to update the fetched
-    // experience based on the cookie here.
-    if (isExperienceClientSideFetched) {
-      debugLog(
-        debug,
-        "Overriding preferences from client-side fetched experience with cookie fides_string consent",
-        cookie.fides_string
-      );
-      const tcfEntities = buildTcfEntitiesFromCookie(experience, cookie);
-      return { cookie, experience: tcfEntities };
-    }
+  if (!isExperienceClientSideFetched) {
     // If it's not client side fetched, we don't update anything since the cookie has already
     // been updated earlier.
     return { cookie, experience };
   }
```

Comment:
hm okay if we don't need the experience to update the cookie anymore, then I don't think we actually need this line anymore, do we? in other words, this function doesn't really care about `isExperienceClientSideFetched`—this function would still return the right thing even without this line (since a server side fetched exp wouldn't have preferences)? we could leave it since it might make things easier to reason about, but it would also be nice to cut out a parameter to this function

Reply:
We still will need the experience to update the cookie, if we don't have a `fides_string` on the `cookie`, but we have prefs on the `experience`. See updated comment here- https://github.com/ethyca/fides/pull/4405#discussion_r1388472610

Does that make sense?

Reply:
hm okay—this flow is confusing but I think it works!